### PR TITLE
[ANCHOR-311] Remove host_url and review usages of web_auth_domain

### PR DESCRIPTION
**Pull request by Ifropc**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Remove host_url and review usages of web_auth_domain

### Why

Previously, host_url was not used outside of SEP-10. In addition, in some instances web_auth_domain was incorrectly used in place of home_domain

### Known limitations

N/A

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java [hunk 1]
```diff
@@ -326,14 +322,14 @@ String generateSep10Jwt(String challengeXdr, String clientDomain)
...
-            getDomainFromURL(appConfig.getHostUrl()),
+            sep10Config.getHomeDomain(),
             sep10Config.getWebAuthDomain());
     debug("challenge:", challenge);
     long issuedAt = challenge.getTransaction().getTimeBounds().getMinTime();
     Memo memo = challenge.getTransaction().getMemo();
     Sep10Jwt sep10Jwt =
         Sep10Jwt.of(
-            appConfig.getHostUrl() + "/auth",
+            sep10Config.getHomeDomain(),
```
Comment (Ifropc) [inline core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java:332]: @JakeUrban should issuer be home domain or web auth domain?

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 2]
```diff
@@ -10,12 +10,10 @@ version: 1
...
 
-## @param: host_url
-## @type: string
-## @supported_prefixes: http:, https:
-## The anchor platform host url.
+
+
 #
-host_url: http://localhost:8080
+host_url:
```
Comment (JakeUrban) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:16]: We can delete this `host_url` attribute can't we?

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 3]
```diff
@@ -251,9 +249,13 @@ sep10:
...
   ## @type: string
   ## The `web_auth_domain` property of SEP-10. https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md#response
-  ## If the `web_auth_domain` is not specified, the `web_auth_domain` will be set to the domain of the value of the `host_url`.
+  ## If the `web_auth_domain` is not specified, the `web_auth_domain` will be set to the domain of the value of the `home_domain`.
   #
   web_auth_domain:
+  ## @param: home_domain
+  ## @type: string
+  ## The `home_domain` property of SEP-10. https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md#request
+  home_domain: localhost:8080
```
Comment (JakeUrban) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:258]: Can we clarify that this should be the host of the toml file? The business can technically host their toml file somewhere else. And if the toml file is not hosted by the platform, the business needs to configure `web_auth_domain` to be the host of the SEP server.

**Review by JakeUrban — APPROVED** [review]
LGTM after the comments are addressed.

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 3 above)
Comment (Ifropc) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:258]: Good idea. We will need to update stellar-docs too with this information

File: platform/src/main/java/org/stellar/anchor/platform/config/PropertyAppConfig.java [hunk 4]
```diff
@@ -46,25 +43,14 @@ void validateConfig(AppConfig config, Errors errors) {
...
-
     if (isEmpty(config.getHorizonUrl())) {
       errors.rejectValue(
           "horizonUrl", "horizon-url-empty", "The stellar_network.horizon_url is not defined.");
     } else {
       if (!NetUtil.isUrlValid(config.getHorizonUrl())) {
         errors.rejectValue(
             "horizonUrl",
-            "horizon-url-invaliad",
+            "horizon-url-invalid",
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/config/PropertyAppConfig.java:53]: 👍

**Review by lijamie98 — APPROVED** [review]
LGTM

**Pull request merged**